### PR TITLE
Add failing test for retrying on status code 503

### PR DESCRIPTION
**Pull request by petermetz**
…ng retried (test is currently failing which is the point)

### CODE REVIEW

File: test/retry.js [hunk 1]
```diff
@@ -28,11 +29,21 @@ test.before('setup', async () => {
...
-});
+	s.on('request', (req, res) => {
+		if (req.method === 'POST' && req.url.includes('fifthPOST')) {
+			console.log(`Sending back HTTP 503 Service Unavailable ...`);
+			fifthPOST++;
+			if (fifthPOST === 5) {
+				res.end(`All is well now, calm down.`);
+			} else {
+				res.statusCode = 503;
+				res.statusMessage = 'Service Unavailable';
```
Comment (sindresorhus) [inline test/retry.js]: This is not needed. It will be set for us by Node.

File: test/retry.js [hunk 2]
```diff
@@ -28,11 +29,21 @@ test.before('setup', async () => {
 		}
 	});
 
-	await s.listen(s.port);
-});
+	s.on('request', (req, res) => {
+		if (req.method === 'POST' && req.url.includes('fifthPOST')) {
+			console.log(`Sending back HTTP 503 Service Unavailable ...`);
```
Comment (sindresorhus) [inline test/retry.js]: Remove this

File: test/retry.js [hunk 3]
```diff
@@ -28,11 +29,21 @@ test.before('setup', async () => {
...
+			fifthPOST++;
+			if (fifthPOST === 5) {
+				res.end(`All is well now, calm down.`);
+			} else {
+				res.statusCode = 503;
+				res.statusMessage = 'Service Unavailable';
+				res.end('You are cool, just make sure you retry this request....');
```
Comment (sindresorhus) [inline test/retry.js]: Use something shorter, like `Error`.

File: test/retry.js [hunk 4]
```diff
@@ -28,11 +29,21 @@ test.before('setup', async () => {
 		}
 	});
 
-	await s.listen(s.port);
-});
+	s.on('request', (req, res) => {
```
Comment (sindresorhus) [inline test/retry.js]: Use `s.on('/retry-503', (req, res) => {`

**Comment by petermetz** [comment]
> Can you change this to use GET? Retrying on POST by default is not safe, so even when we support retrying on HTTP status codes, we won't do POST by default.

This should've been more clear in my PR explanation: I'm not proposing to change the default behavior. I do propose that the documented feature of the `iter` callback to be operational for `POST` verbs as it is for get. This in my opinion doesn't make `POST` retries the default, but it makes it possible for people to opt-in.
My use-case for this was about writing integration tests and verifying some error handling logic on the server side (can't remember the exact details to be honest).

So with all that said my answer is that if you don't approve of the opt-in `POST` retries idea, then the whole PR is pointless anyway and I will rest my case by simply closing the PR.

**Comment by sindresorhus** [comment]
I don't like this way of opting in. It should be even more explicit. I have some ideas on how the retry functionality can be significantly improved: https://github.com/sindresorhus/got/issues/417#issuecomment-366602879

**Pull request closed without merging**